src/Confluent.Kafka/Impl/LibRdKafka.cs

```diff
@@ -766,24 +790,21 @@ private class NativeMethods
             internal static extern ErrorCode rd_kafka_position(
                     IntPtr rk, IntPtr partitions);
 
+            // note: producev signature is rd_kafka_producev(rk, ...)
+            // we are keeping things simple with one binding for now, but it 
+            // will be worth benchmarking the overload with no timestamp, opaque,
+            // partition, etc
             [DllImport(DllName, CallingConvention = CallingConvention.Cdecl)]
-            internal static extern IntPtr rd_kafka_produce(
-                    IntPtr rkt,
-                    int partition,
-                    IntPtr msgflags,
-                    IntPtr val, UIntPtr len,
-                    IntPtr key, UIntPtr keylen,
-                    IntPtr msg_opaque);
-
-            [DllImport(DllName, CallingConvention = CallingConvention.Cdecl)]
-            internal static extern IntPtr rd_kafka_producev(
-                IntPtr rkt,
-                int partition,
-                IntPtr msgflags,
-                IntPtr val, UIntPtr len,
-                IntPtr key, UIntPtr keylen,
-                long timestamp,
-                IntPtr msg_opaque);
+            internal static extern ErrorCode rd_kafka_producev(
+                IntPtr rk,
+                ProduceVarTag topicType, [MarshalAs(UnmanagedType.LPStr)] string topic,
```

Comment:
it would be better if topic name was encoded as UTF-8, however we can leave this for a later PR (same comment applies elsewhere in the lib).